openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -689,3 +693,56 @@ def delete(self, request, usage_key_str, file_path):
         except ValueError:
             raise ValidationError("Invalid file path")  # lint-amnesty, pylint: disable=raise-missing-from
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+@view_auth_classes()
+class LibraryImportTaskViewSet(ViewSet):
+    """
+    Import blocks from Courseware through modulestore.
+    """
+
+    @convert_exceptions
+    def list(self, request, lib_key_str):
```

Comment:
> @jvdm I may be wrong, but it seems to me that the `list` handler does not return ongoing imports.

Is that from a local devstack testing you are doing? Or is it based on the code you are reading?

> Also, how do you feel about removing the `_str` suffix from the `lib_key_str` variable's name?

I am just following the convention on the source code. It's not that bad, since locally, in the scope of the view and other code that manipulate keys, we need a token to help us disambiguate when dealing with the actual key or its string representation.

> Although that's a string, it would be better to indicate its type using type hints instead, like
> 
> ```python
> # from django.http.request import HttpRequest
> 
> def list(self, request: HttpRequest, lib_key: str):
> ```

Unless we have static type checking, I don't think that would help us at all, would it? Reading the code, I would not get the convenience of knowing `lib_key` is the key object or the string object. Please, notice that I am not suggesting we should add type prefixes to all variables from now on, this is a particular case where it seems to make sense to me.